providers/unleash/pkg/provider_test.go

```diff
@@ -0,0 +1,99 @@
+package unleash_test
```

Comment:
I'd really like to see some tests that check that the context passed in is respected. In the test data you have a "users-flag", which would need a context with userid on it to successfully return true. That might be a nice flag to build some tests off of for this

Reply:
[Added](https://github.com/open-feature/go-sdk-contrib/pull/338/files#diff-1cb45209a551a2fbaa8ebc0e5d4be5ae1f5e3381048e165e0ff15c22034b21c4R143).